icaruscode/IcarusObj/PMTBeamSignal.h

```diff
@@ -0,0 +1,57 @@
+/**
+ * @file   icaruscode/IcarusObj/PMTBeamSignal.h
+ * @brief  Holds the event-by-event RWM or EW times
+ * @author Matteo Vicenzi (mvicenzi@bnl.gov)
+ * @date   March 14 2024
+ */
+
+#ifndef ICARUSCODE_ICARUSOBJ_PMTBEAMSIGNAL_H
+#define ICARUSCODE_ICARUSOBJ_PMTBEAMSIGNAL_H
+
+// C/C++ standard libraries
+#include <limits>
+
+namespace icarus::timing{
+		
+        /// Special value to denote no special channel information
+        static constexpr auto NoChannel = std::numeric_limits<unsigned int>::max();
+        /// Special value to denote no time channel information
+        static constexpr double NoTime = 0.;
+        /// Special value to denote no sample information
+        static constexpr size_t NoSample = 0;
+
+	struct PMTBeamSignal {
+		
+		/// Special channel this time was extracted from
+		unsigned int specialChannel = NoChannel;
+	 	
+                /// Board on which the special channel is on
+                std::string digitizerLabel = "";    	
+	 	
+                /// Crate this time applies to
+                std::string crate = "";    
+		
+                /// Sample within the waveform where the reference signal is found
+		size_t sample = NoSample;
+		
+                /// Start time in electronics time [us]
+		double startTimeAbs = NoTime;
+		
+                /// Start time relative to trigger time [us]
+		double startTime = NoTime;
+		
+                PMTBeamSignal(unsigned int ch, std::string b, std::string c,
+			      size_t s, double t, double tt):
+			     specialChannel(ch), digitizerLabel(b), crate(c), sample(s),
+			     startTimeAbs(t), startTime(tt) {};
+                
+                PMTBeamSignal(){};
```

Comment:
I recommend removing all these constructors (partly shown in [`[CF-15]`](https://sbnsoftware.github.io/sbn/codingconv/CodingConventionsList.html#CF157)).
```suggestion
```
Initialisation like
```cpp
  icarus::timing::PMTBeamSignal beamTime{ 0x7063, "EW-TOP-C", "EW-TOP", 13000, -0.02, 1499.98 };
```
will still work via [aggregate initialization](https://en.cppreference.com/w/cpp/language/aggregate_initialization), an the default constructor is implicitly generated.

Reply:
Done!